Windows 1809 Support Alpha

thxCode opened the pull request:
- Adjust tabs to spaces in Dockerfiles
- Support Windows 1809
- Upgrade Flannel & CNI
- Adjust scripts
- Remove Nginx proxy

**Issues**:
https://github.com/rancher/rancher/issues/19173

Commit:
Update flannel-cni to 0.0.2

- Update flanneld binary
    + https://github.com/thxCode/coreos-flannel/releases/tag/v0.2.0-rancher
- Update cni binary
    + https://github.com/thxCode/containernetworking-plugins/releases/tag/v0.2.0-rancher

**Issue:**
https://github.com/rancher/rancher/issues/16460

loganhz commented on VERSION:
```diff
@@ -10,9 +10,6 @@
-    "nginx-proxy": [
-        "v0.0.1"
+        "v0.0.2"
     ]
```
Do we still need to remove this if we don't remove nginx proxy in old version?

thxCode replied:
I am going to use host running nginx instead of container running to avoid https://github.com/rancher/rancher/issues/16074

loganhz commented:
LGTM

loganhz commented:
As this is a breaking change, we need to checkout current master into 2.2 branch first

thxCode commented:
@loganhz , this will not block the releasing for `1803`. After merging this, we can setup an `1809` CI environment with Hyper-V, and then execute `hooks/build.ps1` with `-PowerShellVersion nanoserver-1803`, for example:
```powershell
cd kubernetes-windows-dockerfiles
./hooks/build.ps1 -PowerShellVersion nanoserver-1803
```

The above script can package `1803` with Hyper-V container. The new built packages will have:
```
hyperkube:vx.y.z-nanoserver-1803
flannel-cni:v0.2.0-nanoserver-1803
```

orangedeng commented on scripts/image-clean.ps1:
```diff
@@ -1,5 +1,8 @@
 Write-Host ('Remove all Docker containers ...')
 docker rm -f $(docker ps -aq);
 
+Write-Host ('Prune all container data ...')
```
When do we use this script. It seems very dangerous.

thxCode replied:
There are some tools in `scripts` dir. They are only used for testing.

orangedeng commented:
LGTM

Commit:
Adjust tool scripts

- Adjust `git-install.ps1` and `image-clean.ps1`
- Add `docker-install.ps1`

Commit:
Adjust tabs to spaces in Dockerfiles

- flannel-cni
- hyperkube
- nginx-proxy

Commit:
Remove nginx-proxy

Commit:
Support Windows Server version 1809

- Scripts are using 1809 as default value
- Use `-windows-1809` as image suffix
- Only support Kubernetes 1.14 and above on 1809

thxCode commented:
Fixing is including in https://github.com/rancher/kubernetes-windows-dockerfiles/pull/11, close it.
